### ISSUE TITLE
How to get the Device in ESPHOME / Home Assistant

### ISSUE THREAD

**Issue by f45tb00t** [M4GNV5/esphome-SK20-Nebula-Light#12]
Hi,

everything is flashed on works pretty well. Actually I did some research how to get the device now in ESPHOME and Home Assistant. But unfortunately I did not find that much information about that.

Any help in the right direction would be really nice. 

Thanks in advance.

**Comment by helgek** [comment]
Congrats!
You'll get it into esphome by creating whatever device in esphome and overwriting the yaml content with the yaml configuration you used to flash your projector with. Then the device should show as "online" and future config changes you can flash from inside esphome in home assistant

**Comment by f45tb00t** [comment]
Where can I find the completel yaml which has been flashed? 

only yaml files I can find:
recommended_base.yaml
secrets.yaml

**Comment by helgek** [comment]
Did your device connect to your network, did you provide wifi credentials either through your home assistant yaml or manually through your own yaml configuration?

**Comment by f45tb00t** [comment]
The device is connected to the network. I added the wifi ssid, password, wifi_fallback_password, ota and api password in the secrets.yaml as described
I can reach the interface via http in my wifi

**Comment by helgek** [comment]
This is my code which you can adapt, just double check if the platform and board are correct for you too, but I guess they should be if you were able to use the same config from M4GNV5:

```
substitutions:
  device_name: "Star Projector"
  #store_scene_to_flash: "false"
  friendly_name: Star Projector Bedroom
# Modell SK20

# by default controlling is done via the button and via the web interface
# if you use homeassistant you can uncomment below code:
#api:
#  password: !secret api_password

packages:
  remote_package:
    url: https://github.com/M4GNV5/esphome-SK20-Nebula-Light.git
    ref: master
    #files: [recommended_base.yaml, platform_bk72xx.yaml, nebula_light_device.yaml]
    files: [nebula_light_device.yaml]

esphome:
  name: star-projector-bedroom-device
  friendly_name: ${friendly_name}
  #on_boot:
   # - lambda: |-
    #    id(my_ota).set_auth_password("");
bk72xx:
  board: wb3s

#wifi:
#  ssid: !secret wifi_ssid
#  password: !secret wifi_password
#  power_save_mode: none

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password
  power_save_mode: none
  manual_ip:
    # Set this to the IP of the ESP
    static_ip: 192.168.0.87
    gateway: 192.168.0.1
    subnet: 255.255.255.0

  ap:
    ssid: "Fallback AP"
    password: !secret wifi_fallback_password

captive_portal:

logger:
  level: INFO

ota:
  - platform: esphome
    password: !secret ota_password
  #password: !secret ota_password
  #id: my_ota

web_server:
  port: 80

api:

```

**Comment by f45tb00t** [comment]
Thanks a lot for the config. But seems it does not work. I just added the needed stuff like ssid, password, .... but still offline. 

I tried to check the logs and get the following
```
INFO ESPHome 2024.11.1
INFO Reading configuration /config/nebula1.yaml...
INFO Starting log output from 10.* using esphome API
WARNING Can't connect to ESPHome API for star-projector-bedroom-device @ 10.*: Error connecting to [AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='10.*, port=6053))]: [Errno 111] Connect call failed ('10.*, 6053) (SocketAPIError)
INFO Trying to connect to star-projector-bedroom-device @ 10.* in the background
```

But that makes sense. Because the port 6053 is not open when you use the guide from M4GNV5. 

I did not do that much with esphome so far. 
Is it possible to use the webgui on port 80 of the device itself to upload that configuration? Like create a text file, copy and paste, save as configuration.yaml and just upload?

Edit: 

Only open ports I have:
Discovered open port 80/tcp on 10.*
Discovered open port 8892/tcp on 10.*

**Comment by helgek** [comment]
With ports I don’t know, I never had to deal with this using my 20+ ESPHome based devices in my network. Normally if the IP address is set correctly it should find the device. And Home Assistant should also normally discover your device regardless if it is set up in the ESPHome dashboard or not. 

I don’t remember if the configuration can be uploaded through the webserver ui but anyway it shouldn’t be required since you only want the device to show up in ESPHome. For this you don’t even need to reflash the device. You’re just creating a dummy yaml device in ESPHome with the correct ip setting and then ESPHome should have access to it.

**Comment by f45tb00t** [comment]
Okay.... I have it. With the help of you and  MrRodz in the HA Forum.

For others who have the same issue.

1. Create a new random device in esphome 
2. use the input of recommended_base.yaml or the config of helgek (I used helgeks), fill in this needed fields (SSID, passwords for wifi, ota, ....) and replace the random device config. Make sure you don't have dupe entries.
3. click three dots of the device in esphome. Then click validate. This should give "INFO Configuration is valid!" After that click Install - Wirelessly. This  will install everything via esphome and the device will be online afterwards.


Thanks a lot guys. There is always a first time to do things :)


By the way. It seems the tools did not like my OS and Wifi Adapters in VMs.  I tried with Ubuntu(issues with apparmor), Debian and Kali Linux. I ended up just to use a fresh Debian  Installation on my Laptop with the internal Wifi chip. This worked directly for me.
